modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorCache.ts

```diff
@@ -1,16 +1,21 @@
-import { Arr, Type } from '@ephox/katamari';
+import { Arr, Obj, Type } from '@ephox/katamari';
 
+import { Menu } from 'tinymce/core/api/ui/Ui';
 import LocalStorage from 'tinymce/core/api/util/LocalStorage';
 
-const storageName = 'tinymce-custom-colors';
-
 export interface ColorCache {
   readonly add: (key: string) => void;
   readonly state: () => string[];
 }
 
-export const ColorCache = (max: number = 10): ColorCache => {
-  const storageString = LocalStorage.getItem(storageName);
+interface CacheStorage {
+  [index: string]: ColorCache;
+}
+const cacheStorage: CacheStorage = {};
+
+const ColorCache = (id: string, max: number = 10): ColorCache => {
+  const storageId = `tinymce-custom-colors-${id}`;
```

Comment:
This is a **breaking** change for end users so cannot be done... it needs to continue to use `tinymce-custom-colors` as the default key so that users custom colors continue to work when they upgrade.

Reply:
Yes I think prefer the new values, but if they don't exist read the old one and copy it no matter what the ID is. That way the value carries forward to both but can then diverge.